e2e/cypress/tests/beforeLoginTests.spec.ts

```diff
@@ -0,0 +1,19 @@
+//test ability to browse/download codebases
+describe("Browse Codebases", () => {
+  it("should browse the model library and be able to enter a search query", () => {
+    cy.visit("/"); // visit homepage
+    cy.visit("/codebases");
+    assert(cy.get("h1").contains("Computational Model Library"));
+  });
+});
+
+//download codebase
+describe("Download Codebases", () => {
+  it("should download a codebase", () => {
+    // Navigate to the first 'search-result', find the 'a' tag within the 'title' div, and click it
+    cy.get(".search-result", { timeout: 10000 })
+      .first()
+      .find(".title a")
```

Comment:
this filtering criteria seems poorly vague for identifying a codebase download. Setup a more precise criteria even if you need to include additional metadata on the model library page.